runtime/src/chpl-visual-debug.c

```diff
@@ -51,6 +51,11 @@ static void cb_task_end(const chpl_task_cb_info_t *info);
 
 int chpl_vdebug_fd = -1;
 int chpl_vdebug = 0;
+int chpl_vdebug_task = 0;
```

Comment:
I think you still have a problem with this. You've got a global variable here, unprotected by a lock, which is subject to data races (conflicting assignments) if more than one task calls chpl_vdebug_getTid() at the same time.